api/v1/mongodbcommunity_types.go

```diff
@@ -339,8 +352,24 @@ func (m MongodConfiguration) GetDBDataDir() string {
 // GetDBPort returns the port that should be used for the mongod process.
 // If port is not specified, the default port of 27017 will be used.
 func (m MongodConfiguration) GetDBPort() int {
-	// When passed as a number "net.port" is unmarshalled into a float64
-	return int(objx.New(m.Object).Get("net.port").Float64(float64(automationconfig.DefaultDBPort)))
+	portValue := objx.New(m.Object).Get("net.port")
+
+	// Underlying map could be manipulated in memory, e.g. via SetDBPort - then it will be as int,
+	// or it could be deserialized from JSON and then integer in an untyped map will be deserialized as float64.
+	// It's behavior of https://pkg.go.dev/encoding/json#Unmarshal that is converting JSON integers as float64.
+	if portValue.IsInt() {
+		return portValue.Int(automationconfig.DefaultDBPort)
+	} else if portValue.IsFloat64() {
+		return int(portValue.Float64(float64(automationconfig.DefaultDBPort)))
+	}
+
+	return automationconfig.DefaultDBPort
```

Comment:
Yes, definitely. But now it's not easy to log from inside these structures. So probably we would need to return error here and this would be an API change. So I decided to just be backwards compatible.
I would rather opt for validation routine which would check if everything is OK and not validate ad hoc in getters.

Reply:
This is actually a nice idea! I like it!